### PR TITLE
Add solution for 2017-day4

### LINKED ISSUE #57: Solve 2017 Day 4

### PR DESCRIPTION
**Pull request by Glyphack**
Closes https://github.com/Kushagra-0801/Advent-Of-Code/issues/57

### CODE REVIEW

File: 2017/Day 4 - 1.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+unique_passphrases_count = 0
+passphrases = []
+with open('Day 4 - input', 'r') as f:
+    for line in f:
+        passphrases.append(list(line.split()))
+    for passphrase in passphrases:
```
Comment (Kushagra-0801) [inline 2017/Day 4 - 1.py]: no need to store all the passphrases in a list
Just process them as they come

File: 2017/Day 4 - 1.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+        if len(passphrase) == len(set(passphrase)):
+            unique_passphrases_count += 1
+print(unique_passphrases_count)
```
Comment (Kushagra-0801) [inline 2017/Day 4 - 1.py]: Add a new line

File: 2017/Day 4 - 2.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import itertools
+
+valid_passphrases_count = 0
+passphrases = []
+
+with open('Day 4 - input', 'r') as f:
+    for line in f:
+        passphrases.append(list(line.split()))
```
Comment (Kushagra-0801) [inline 2017/Day 4 - 2.py]: Same thing. No need for this

File: 2017/Day 4 - 2.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import itertools
```
Comment (Kushagra-0801) [inline 2017/Day 4 - 2.py]: unused import

File: 2017/Day 4 - 2.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+valid_passphrases_count = 0
+passphrases = []
+
+with open('Day 4 - input', 'r') as f:
+    for line in f:
+        passphrases.append(list(line.split()))
+    for passphrase_list in passphrases:
+        passphrase = list(map(lambda x: ('').join(sorted(list(x))), passphrase_list))
```
Comment (Kushagra-0801) [inline 2017/Day 4 - 2.py]: It is preferable to use a `list comprehension` than calling `list` on `map`.
Also, sorted works fine on strings, no need to convert it to a list. Though the return object is a list and needs to be joined

File: 2017/Day 4 - 2.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+        if len(passphrase) == len(set(passphrase)):
+            valid_passphrases_count += 1
+        
+
```
Comment (Kushagra-0801) [inline 2017/Day 4 - 2.py]: Unnecessary vertical whitespace

**Review by Kushagra-0801 — CHANGES_REQUESTED** [review]
Please make the given changes

**Review by namannimmo10 — APPROVED** [review]
LGTM @Kushagra-0801. Thanks for your contribution. @Glyphack

**Pull request merged**